config/webhook/additional_secret_hooks.yaml

```diff
@@ -0,0 +1,58 @@
+# https://github.com/kubernetes-sigs/controller-tools/issues/553
+# controller-tools, and by extension kubebuilder, do not support specifying objectSelector,
+# which we need for the Secret rules.
+apiVersion: admissionregistration.k8s.io/v1
+kind: ValidatingWebhookConfiguration
+metadata:
+  name: validating-webhook-configuration
+webhooks:
+- admissionReviewVersions:
+  - v1
+  clientConfig:
+    service:
+      name: webhook-service
+      namespace: system
+      path: /
+  failurePolicy: Fail
+  matchPolicy: Equivalent
+  name: secrets.credentials.validation.ingress-controller.konghq.com
+  objectSelector:
+    matchExpressions:
+    - key: "konghq.com/credential"
+      operator: "Exists"
+  rules:
+  - apiGroups:
+    - ""
+    apiVersions:
+    - v1
+    operations:
+    - CREATE
+    - UPDATE
+    resources:
+    - secrets
+  sideEffects: None
+- admissionReviewVersions:
+  - v1
+  clientConfig:
+    service:
+      name: webhook-service
+      namespace: system
+      path: /
+  failurePolicy: Fail
+  matchPolicy: Equivalent
+  name: secrets.plugins.validation.ingress-controller.konghq.com
+  objectSelector:
+    matchExpressions:
+    - key: "konghq.com/validate"
+      operator: "Exists"
```

Comment:
I'm curious what would be the meaning of this value? That the checked `Secret` is part of a plugin configuration?

---

Do we have an idea of other values that we'd allow here?

Reply:
Yeah, my initial proposal was to simply go with `konghq.com/validate: "true"` which is easier to understand (please validate this Secret for me). As I understand @rainest proposed using specific values to be able to distinguish between potential future Secret "kinds" so the validation webhook can dispatch based on that to different validation branches (for now it doesn't matter though, we dispatch based on the Secret's referrer; I can imagine that we may have something similar to credentials in the future that would be validated with no referrer - in such case we could add another label similar to `konghq.com/credential` to make the distinction).  

> Do we have an idea of other values that we'd allow here?
Not really, @rainest?

Reply:
My thinking was to just check for `konghq.com/validate` in the manifest (to limit the number of clauses if we expand it) and then use the value to differentiate behavior in the handler. The current `default` handling may not be appropriate for everything, but is fine ignoring things it does not recognize if someone happened to set `konghq.com/validate: ldjgdf` on some irrelevant Secret.

Separating clauses in the manifest doesn't get us anything in the manifest since they're all ultimately dumped into the same handler.

> Do we have an idea of other values that we'd allow here?

Greg is correct, there isn't any known one. I chose the generic key so we wouldn't have to add another Secret label later if we do need it, but a specific value because some future validation may not be able to use a heuristic like the existing "if referrer exists and is KongPlugin, do plugin validation".

Offhand the next thing I can think of is custom entities, which would presumably support Secret references (they're often credentials). However, those _can_ use the referrer heuristic.

Reply:
Well, yeah, probably it won't hurt us to do it this way and will give us the possibility of adding more Secrets flavors. I'm fine with going with `kong.com/validate: secret-kind` and leaving the webhook config as proposed (filter on presence of the label, not its values).